nixos/modules/installer/tools/tools.nix

```diff
@@ -224,12 +224,22 @@ in
         # accidentally delete configuration.nix.
         # system.copySystemConfiguration = true;
 
-        # This value determines the NixOS release from which the default
-        # settings for stateful data, like file locations and database versions
-        # on your system were taken. It's perfectly fine and recommended to leave
-        # this value at the release version of the first install of this system.
-        # Before changing this value read the documentation for this option
-        # (e.g. man configuration.nix or on https://nixos.org/nixos/options.html).
+        # This option defines the first version of NixOS you have installed on this particular machine, and is
+        # used to maintain compatibility with application data (e.g. databases) created on older NixOS versions.
+        #
+        # Most users should NEVER change this value after the initial install, for any reason,
+        # even if you've upgraded your system.
```

Comment:
The one issue I have with this wording is that "upgrade" is ambiguous between a mere `nix-channel --update`, and a *release upgrade*, in which you switch which channel you are following. I'd like to emphasize that we are specifically talking about release upgrades.

Reply:
It applies to either, though?

Reply:
It does, but I keep seeing people think "but release upgrades are special, right?", so I think it's best to avoid that misunderstanding. Also, given the values this option takes, people shouldn't think they needed to change it for normal updates anyway, after all, what would they be changing it to in that case?

Reply:
so maybe just
```suggestion
        # even if you've upgraded your system to a new nixos release.
```

Reply:
I guess.